server/filter.js

```diff
@@ -344,12 +344,26 @@ function parseQuery (query) {
   return expression;
 }
 
-function filterQuery(items, query) {
-  var expression = parseQuery(query);
-  var result = items.filter(function(item) {
-    return expression.matches(item);
-  });
-  return result;
+function filterQuery(items, query, filter) {
```

Comment:
Since query and filter are orthogonal, I think it would be a little cleaner to have a separate function (filterFilter()?) rather than putting both inside filterQuery.